meeteval/wer/wer/time_constrained.py

```diff
@@ -351,24 +389,24 @@ def get_pseudo_word_level_timings(
     """
     pseudo_word_level_strategy = pseudo_word_level_strategies[strategy]
 
-    all_words = []
-    word_level_timings = []
+    def get_words(s):
+        res = []
+        words = s['words'].split()
+        if not words:  # Make sure that we don't drop a speaker
+            words = ['']
+        for w, (start, end) in zip(words, pseudo_word_level_strategy((s['start_time'], s['end_time']), words)):
+            res.append({**s, 'words': w, 'start_time': start, 'end_time': end})
+        return res
 
-    for words, interval in zip(s.transcript, s.timings):
-        words = words.split()  # Get words form segment
-        segment_timings = pseudo_word_level_strategy(interval, words)
-        word_level_timings.extend(segment_timings)
-        all_words.extend(words)
-        assert len(words) == len(segment_timings), (words, segment_timings)
-
-    return TimeMarkedTranscript(all_words, word_level_timings)
+    return t.flatmap(get_words)
 
 
+@seglst_map()
 def remove_overlaps(
-        s: TimeMarkedTranscript,
+        t: SegLST,
         max_overlap: float = 0.4,
         warn_message: str = None,
-) -> TimeMarkedTranscript:
+) -> SegLST:
     """
     Remove overlaps between words or segments in a transcript.
```

Comment:
It it reasonable to sort in the function? I think that's the better solution.

Reply:
I think the initial `TimeMarkedTranscript` was sorted. Now I am not sure, whether the input will be sorted.

> It it reasonable to sort in the function? I think that's the better solution.

It depends. When the sort is not necessary, it may affect the runtime.
When I tried to optimize the tcpWER by mapping floats to ints, the sort was far too expensive to give a runtime improvement.

Reply:
The sort is necessary and the function is currently unused. It because unused when we added the sort arguments to the tcpWER. I would like to keep the function because it is needed when we want to include an asclite wrapper that evaluates in the same way as the original Libri-CSS papers.

Reply:
Ah, ok. Then adding the sort is a good idea.